typescript/infra/src/agents/key-utils.ts

```diff
@@ -399,8 +407,10 @@ async function persistAddressesLocally(
     multisigValidatorKeys[key.chainName] ||= {
       validators: [],
     };
-    if (key.chainName)
+
+    if (key.role === Role.Validator) {
```

Comment:
Can there ever be a case where the `role` is `validator` but `key.chainName` is undefined?

Reply:
no - clarified as well